src/ziggurat/metrics.clj

```diff
@@ -76,22 +81,32 @@
    (inc-or-dec-count sign {:metric-namespace metric-namespace :metric metric :n n :additional-tags additional-tags}))
   ([sign {:keys [metric-namespace metric n additional-tags]}]
    (let [metric-ns (get-metric-namespaces metric-namespace)
-         meter     ^Meter (mk-meter metric-ns metric (get-map additional-tags))]
+         meter     ^Meter (mk-meter metric-ns metric (remove-topic-tag-for-old-namespace (get-map additional-tags) metric-namespace))]
      (.mark meter (sign (get-int n))))))
 
 (def increment-count (partial inc-or-dec-count +))
 
 (def decrement-count (partial inc-or-dec-count -))
 
+(defn multi-ns-increment-count [nss metric additional-tags]
+  (doseq [ns nss]
+    (increment-count ns metric additional-tags)))
+
 (defn report-histogram
   ([metric-namespaces val]
    (report-histogram metric-namespaces val nil))
   ([metric-namespaces val additional-tags]
    (let [metric-namespace (get-metric-namespaces metric-namespaces)
-         histogram        ^Histogram (mk-histogram metric-namespace "all" additional-tags)]
+         histogram        ^Histogram (mk-histogram metric-namespace "all" (remove-topic-tag-for-old-namespace additional-tags metric-namespaces))] ;; verify if get-map is needed here
      (.update histogram (get-int val)))))
 
-(def report-time report-histogram)                         ;; for backward compatibility
+(def report-time report-histogram)                          ;; for backward compatibility
+
+(defn multi-ns-report-histogram [nss time-val additional-tags]
```

Comment:
We should have a test for multi-ns-report-histogram

Reply:
added

Reply:
done